rust/xaynet-server/src/state_machine/phases/error.rs

```diff
@@ -48,31 +63,16 @@ impl Phase for PhaseState<PhaseStateError> {
     const NAME: PhaseName = PhaseName::Error;
 
     async fn run(&mut self) -> Result<(), PhaseStateError> {
-        error!("state failed: {}", self.inner);
+        error!("phase state error: {}", self.inner);
 
         metrics!(
             self.shared.io.metrics_tx,
             metrics::phase::error::emit(&self.inner)
         );
 
-        if let PhaseStateError::Redis(_) = self.inner {
-            // a simple loop that stops as soon as the redis client has reconnected to a redis
-            // instance. Reconnecting a lost connection is handled internally by
-            // redis::aio::ConnectionManager
-
-            while self
-                .shared
-                .io
-                .redis
-                .connection()
-                .await
-                .ping()
-                .await
-                .is_err()
-            {
-                info!("try to reconnect to Redis in 5 sec");
-                delay_for(Duration::from_secs(5)).await;
-            }
+        while self.shared.io.redis.coordinator_state().await.is_err() {
```

Comment:
I would like to improve this part in a follow up pr

Reply:
What kind of improvement are you planning on making here?

Reply:
I'm not really sure yet

One option I see here is to add something like a `health` or `is_ready` method 
- for external storage solution it would return `Ok` once the client was able to reconnect
- for in-mem / filesystem it would always return `Ok` / or we could also check if enough space is left or so 

Another option could be that the memory itself is blocked until for example a client is reconnect. However, this can mean that the state machine is blocking in unusual places.

e.g.

https://github.com/xaynetwork/xaynet/blob/4fcf402a4f7cf2866d071b51bf572a2d4a1a7ba6/rust/xaynet-server/src/state_machine/phases/idle.rs#L40-L71

lets say the coordinator can set the state but it cannot broadcast the round_params because the redis store blocks in the `flush_dicts` method